### PR TITLE
Stats: Introduce Tier Upgrade Notice for showing upgrade information

### PR DESCRIPTION
**Pull request by dognose24**
Related to #84671

## Proposed Changes

* Introduce the `TierUpgradeNotice` to show information for the upgrade flow.

## Testing Instructions

* Spin this change on the local Jetpack environment: `STATS_PACKAGE_PATH=/path-to-jetpack/projects/packages/stats-admin yarn dev`.
* Edit the code to disable `isCommercial` check for a non-commercial testing site. https://github.com/Automattic/wp-calypso/pull/84880/files#diff-bbc3bb485e5fb5d297cfdc0c767f67a313b15896520f130e68ad3292b40cecdbR134
* Navigate to the Stats > `Traffic` page with the feature flag for a site with Stats Commercial license: `/wp-admin/admin.php?page=stats#!/stats/day/{site-id}?page=stats&flags=stats%2Ftier-upgrade-slider`.
* Edit the code with different `currentUsage` to test for different notices. https://github.com/Automattic/wp-calypso/pull/84880/files#diff-f018bb72763ca969cc57532cf3aa3f75c77cff12eb7589d101dd34de956b4ae0R21
* Ensure the Tier Upgrade Notice shows messages according to the current usage.

截圖 2023-12-06 下午11 35 14

截圖 2023-12-06 下午11 35 52

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-notices/all-notice-definitions.ts [hunk 1]
```diff
@@ -105,6 +106,37 @@ const ALL_STATS_NOTICES: StatsNoticeType[] = [
...
+			isWpcom,
+			isVip,
+			isP2,
+			isOwnedByTeam51,
+			hasPaidStats,
+			isSiteJetpackNotAtomic,
+			isCommercial,
+		}: StatsNoticeProps ) => {
+			// TODO: Maybe we won't show tier upgrade notice for WPCOM sites for now?
+			const showTierUpgradeNoticeForWpcomSites = isWpcom && ! isVip && ! isP2 && ! isOwnedByTeam51;
```
Comment (kangzj) [inline client/my-sites/stats/stats-notices/all-notice-definitions.ts:123]: We can stop showing notices for anymore wpcom sites now, as the Explorers team will be working on those.

File: client/my-sites/stats/hooks/use-notice-visibility-query.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ const DEFAULT_SERVER_NOTICES_VISIBILITY = {
 	traffic_page_settings: false,
 	do_you_love_jetpack_stats: false,
 	commercial_site_upgrade: false,
+	// TODO: Check if the site needs to be upgraded to a higher tier on the back end.
+	tier_upgrade_notice: true,
```
Comment (kangzj) [inline client/my-sites/stats/hooks/use-notice-visibility-query.ts:11]: I imagine the data from the usage API would be sufficient? We'll still need to add status support to the notices API tho.

**Review by kangzj — APPROVED** [review]
Thanks for working on this! I'm pretty happy for it to land as is and then add the conditions/usage etc. Good job 👍

File: client/my-sites/stats/hooks/use-notice-visibility-query.ts (hunk 2 above)
Comment (dognose24) [reply client/my-sites/stats/hooks/use-notice-visibility-query.ts:11]: > I imagine the data from the usage API would be sufficient? 

Yes, it's enough for now. I just wanted to note whether it should be tackled as a consolidation for notices. But it is reasonable to be apart as well. 🙂

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10036708

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @dognose24 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10036708).

**Pull request merged**